File: app/src/main/java/wallgram/hd/wallpapers/views/AgreementDialog.java
package wallgram.hd.wallpapers.views;

import android.annotation.TargetApi;
import android.app.Dialog;
import android.os.Build;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

import wallgram.hd.wallpapers.R;
import wallgram.hd.wallpapers.util.Common;

public class AgreementDialog extends DialogFragment {

    private static final String TAG = "agreement_dialog";

    private Toolbar toolbar;
    private WebView webView;
    private ProgressBar progressBar;

    public static void display(FragmentManager fragmentManager) {
        AgreementDialog agreementDialog = new AgreementDialog();
        agreementDialog.show(fragmentManager, TAG);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, R.style.AppTheme_FullScreenDialog);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(R.layout.agreement_dialog, container, false);

        toolbar = view.findViewById(R.id.toolbar);
        webView = view.findViewById(R.id.web_view);
        progressBar = view.findViewById(R.id.progress_bar);

        loadPage();

        return view;
    }

    private void loadPage() {
        webView.loadUrl(Common.getSiteUrl() + "privacy");
        webView.setWebViewClient(webViewClient);
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        toolbar.setNavigationOnClickListener(v -> dismiss());
        toolbar.setTitle(R.string.privacy_policy);
        toolbar.setOnMenuItemClickListener(item -> {
            dismiss();
            return true;
        });
    }

    @Override
    public void onStart() {
        super.onStart();
        Dialog dialog = getDialog();
        if (dialog != null) {
            int width = ViewGroup.LayoutParams.MATCH_PARENT;
            int height = ViewGroup.LayoutParams.MATCH_PARENT;
            if(dialog.getWindow() != null){
                dialog.getWindow().setLayout(width, height);
                dialog.getWindow().setWindowAnimations(R.style.AppTheme_Slide);
            }
        }
    }

    private final WebViewClient webViewClient = new WebViewClient() {
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            view.loadUrl(url);
            return true;
        }

        @TargetApi(Build.VERSION_CODES.N) @Override
        public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
            view.loadUrl(request.getUrl().toString());
            return true;
        }

        @Override public void onPageFinished(WebView view, String url) {
            super.onPageFinished(view, url);
            if(progressBar != null)
                progressBar.setVisibility(View.GONE);
        }
    };
}

File: app/src/main/java/wallgram/hd/wallpapers/views/AgreementDialog.kt
package wallgram.hd.wallpapers.views

import android.webkit.WebView
import android.os.Bundle
import wallgram.hd.wallpapers.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.webkit.WebViewClient
import android.annotation.TargetApi
import android.os.Build
import android.view.View
import android.webkit.WebResourceRequest
import androidx.core.view.isVisible
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import wallgram.hd.wallpapers.databinding.AgreementDialogBinding
import wallgram.hd.wallpapers.util.Common

class AgreementDialog : DialogFragment() {

    private var binding: AgreementDialogBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NORMAL, R.style.AppTheme_FullScreenDialog)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = AgreementDialogBinding.inflate(inflater, container, false)

        return binding?.root
    }

    private fun loadPage() {
        binding?.apply {
            webView.apply {
                loadUrl(Common.getSiteUrl() + "terms")
                webViewClient = mWebViewClient
            }
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        loadPage()

        binding?.apply {
            toolbar.apply {
                setNavigationOnClickListener {
                    dismiss()
                }
                setTitle(R.string.privacy_policy)
                setOnMenuItemClickListener {
                    dismiss()
                    true
                }
            }
        }
    }

    override fun onStart() {
        super.onStart()
        val dialog = dialog
        if (dialog != null) {
            val width = ViewGroup.LayoutParams.MATCH_PARENT
            val height = ViewGroup.LayoutParams.MATCH_PARENT
            if (dialog.window != null) {
                dialog.window!!.setLayout(width, height)
                dialog.window!!.setWindowAnimations(R.style.AppTheme_Slide)
            }
        }
    }

    private val mWebViewClient: WebViewClient = object : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            view.loadUrl(url)
            return true
        }

        @TargetApi(Build.VERSION_CODES.N)
        override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
            view.loadUrl(request.url.toString())
            return true
        }

        override fun onPageFinished(view: WebView, url: String) {
            super.onPageFinished(view, url)
            binding?.apply {
                progressBar.isVisible = false
            }
        }
    }

    companion object {
        private const val TAG = "agreement_dialog"
        fun display(fragmentManager: FragmentManager) {
            val agreementDialog = AgreementDialog()
            agreementDialog.show(fragmentManager, TAG)
        }
    }
}